cmd/containerd-shim-runhcs-v1/task_hcs.go

```diff
@@ -961,3 +972,95 @@ func (ht *hcsTask) ProcessorInfo(ctx context.Context) (*processorInfo, error) {
 		count: ht.host.ProcessorCount(),
 	}, nil
 }
+
+// Add mount as vSMB share to the UVM at the given destination path
+func (ht *hcsTask) addMountToUVM(ctx context.Context, src string, dst string, isRO bool) (string, error) {
+	options := ht.host.DefaultVSMBOptions(isRO)
+	vsmbShare, err := ht.host.AddVSMB(ctx, src, options)
+	if err != nil {
+		return "", errors.Wrapf(err, "failed to add mount as vSMB share to UVM")
+	}
+	defer func() {
+		if err != nil {
+			_ = vsmbShare.Release(ctx)
+		}
+	}()
+
+	sharePath, err := ht.host.GetVSMBUvmPath(ctx, src, isRO)
+	if err != nil {
+		return "", errors.Wrapf(err, "failed to get vsmb path")
+	}
+	return sharePath, nil
+}
+
+func (ht *hcsTask) requestAddContainerMount(ctx context.Context, resourcePath string, settings interface{}) error {
+	modification := &hcsschema.ModifySettingRequest{
+		ResourcePath: resourcePath,
+		RequestType:  guestrequest.RequestTypeAdd,
+		Settings:     settings,
+	}
+	return ht.c.Modify(ctx, modification)
+}
+
+func isMountTypeSupported(mountType string) bool {
+	// currently we only support mounting of host volumes/directories
+	return mountType == ""
+}
+
+func (ht *hcsTask) updateWCOWContainerMount(ctx context.Context, data interface{}, annotations map[string]string) error {
+	// Hcsschema v2 should be supported
+	if osversion.Build() >= osversion.RS5 {
+		if sv := schemaversion.DetermineSchemaVersion(nil); !schemaversion.IsV21(sv) {
+			return fmt.Errorf("hcsschema v2 unsupported")
+		}
+	}
+	resources, ok := data.(*ctrdtaskapi.ContainerMount)
+	if !ok {
+		return errors.New("resources must be of type *ctrdtaskapi.ContainerMount when adding mount to a running windows container")
+	}
+	if resources.HostPath == "" || resources.ContainerPath == "" {
+		return fmt.Errorf("invalid OCI spec - a mount must have both host and container path set")
+	}
+
+	// Check for valid mount type
+	if !isMountTypeSupported(resources.Type) {
+		return fmt.Errorf("invalid mount type - currently only host volumes/directories can be mounted to running containers")
+	}
+
+	if ht.host == nil {
+		// HCS has a bug where it does not correctly resolve file (not dir) paths
+		// if the path includes a symlink. Therefore, we resolve the path here before
+		// passing it in. The issue does not occur with VSMB, so don't need to worry
+		// about the isolated case.
```

Comment:
Doe this still exist today? Maybe it was fixed later?

Reply:
I still see this in the create container document code path during container creation so it probably still exists or isn't cleaned up from the codebase?

Reply:
If the src path is passed without resolving the path in cri and hcsshim there seems to be an issue with mounting still - so I think its probably still around

Reply:
Can we make sure there is an OS bug opened for this?

Reply:
opened one for this